### PR TITLE
#213 Admin page

### PR DESCRIPTION
**Pull request by longtimeago**
For #213 Admin page is implemented. For now it's insecure and shows docs of currently logged user only.

### CODE REVIEW

**Comment by longtimeago** [comment]
@karato please, find someone to review the PR

**Comment by karato** [comment]
@longtimeago Many thanks, I will find someone to review it before we merge

**Comment by karato** [comment]
@darkled do this review

**Comment by asaen** [comment]
@rultor merge

File: src/main/java/com/nerodesk/takes/TkAdmin.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    @Override
+    public Response act(final Request req) throws IOException {
+        final User user = this.base.user(new RqAuth(req).identity().urn());
+        return new RsPage(
+            "/xsl/admin.xsl",
+            req,
+            new XeSource() {
+                @Override
+                public Iterable<Directive> toXembly() throws IOException {
+                    return TkAdmin.this.list(user);
```
Comment (yegor256) [inline src/main/java/com/nerodesk/takes/TkAdmin.java]: it's better to use [`XeTransform`](http://www.takes.org/apidocs-0.16.9/org/takes/rs/xe/XeTransform.html)

**Comment by yegor256** [comment]
@longtimeago I would suggest to use `XeTransform` instead, code will be cleaner

**Comment by longtimeago** [comment]
@yegor256 please, see ones again. Hope, you like this approach more than previous one :)

File: src/main/java/com/nerodesk/takes/TkAdmin.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                        @Override
+                        public XeSource transform(final String name) {
+                            return new XeAppend(
+                                "doc",
+                                new XeDirectives(
+                                    new Directives()
+                                        .add("path").set(
+                                        String.format(
+                                            "%s/%s",
+                                            user.urn().replaceAll(":", "/"),
```
Comment (yegor256) [inline src/main/java/com/nerodesk/takes/TkAdmin.java]: @longtimeago what this is for? I didn't get it... we don't store files like that. a real path of a file in S3 is `urn:facebook:1234567/test.txt`

File: src/main/java/com/nerodesk/takes/TkApp.java [hunk 3]
```diff
@@ -185,6 +185,10 @@ public void run() {
                 )
             ),
             new FkRegex(
+                "/admin",
+                new TkSecure(new TkGreedy(new TkAdmin(base)))
```
Comment (yegor256) [inline src/main/java/com/nerodesk/takes/TkApp.java]: @longtimeago TkGreedy is not needed here, it is only required when you are planning to read HTTP request body multiple times

**Comment by yegor256** [comment]
@longtimeago much nicer now :) a few more comments

**Comment by longtimeago** [comment]
@yegor256 one more CR round :)

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by karato** [comment]
@darkled Thanks so much! Your account was topped up for **16 mins** (transaction ID is `57639842`, the task took 49 hours and 23 mins); extra minutes for review comments (c=1); +16 added to your rating, current score is: [+1988](http://www.netbout.com/b/35091?open=rating)

**Comment by karato** [comment]
@rultor please deploy

**Pull request merged**